src/server/data_export_csv.py

```diff
@@ -0,0 +1,319 @@
+'''CSV data exporter'''
+
+import logging
+logger = logging.getLogger(__name__)
+from Language import __
+import RHUtils
+import io
+import csv
+from sqlalchemy.ext.declarative import DeclarativeMeta
+from data_export import DataExporter
+
+def write_csv(data):
+    output = io.StringIO()
+    writer = csv.writer(output, quoting=csv.QUOTE_NONNUMERIC)
+    writer.writerows(data)
+
+    return {
+        'data': output.getvalue(),
+        'encoding': 'text/csv',
+        'ext': 'csv'
+    }
+
+def assemble_all(Database, PageCache):
```

Comment:
This is duplicated in data_export_csv and data_export_json. Let's combine

Reply:
Not perfectly. The JSON 'all' assembler just takes the output and puts them in separate keys. The CSV assembler uses the key as a header and writes the individual outputs data into a flat structure. Each is suitable for the formatting of that type.

There probably could be a single function for both, but the CSV formatter would then also be responsible for flattening the data if it was needed.

Reply:
As part of the export, there should be a clear separation between _getting data from the database in a usable format_ and _outputting it in the exporter_. Thus, the code for getting the data is the same for any exporter and can be shared in DatabaseReader for example.